### PR TITLE
Move Disk to context manager

### PR DESCRIPTION
**Pull request by schaefi**
Change the Disk class to be a context manager. All code using Disk was updated to the following
with statement:

```python
with Disk(...) as disk:
    disk.some_member()
```

This is related to Issue #2412

### CODE REVIEW

**Comment by schaefi** [comment]
next round in moving to context managers

**Review by dcermak — COMMENTED** [review]
LGTM, I have just added a small nit about the systemd-boot code

File: kiwi/bootloader/config/systemd_boot.py [hunk 1]
```diff
@@ -119,46 +119,45 @@ def _create_embedded_fat_efi_image(self, path: str):
...
+                proc_mount.bind_mount()
+                sys_mount.bind_mount()
+                try:
+                    self._run_bootctl(self.root_dir)
+                    self.set_loader_entry(self.root_dir, self.target.live)
+                finally:
+                    efi_mount.umount()
+                    device_mount.umount()
+                    proc_mount.umount()
+                    sys_mount.umount()
```
[the remark this answers is not in the document]
Comment (schaefi) [reply kiwi/bootloader/config/systemd_boot.py:157]: yes good point

[the remark this answers is not in the document]
Comment (schaefi) [reply kiwi/bootloader/config/systemd_boot.py]: done

**Review by dcermak — APPROVED** [review]

**Pull request merged**